### PR TITLE
Implement is_qat in TorchVision

### PR DESCRIPTION
**Pull request by datumbox**
Due to the breaking changes introduced at https://github.com/pytorch/pytorch/pull/70009 and https://github.com/pytorch/pytorch/pull/71956, TorchVision needs to make changes on the quantization code. This PR should make adjustments in a BC manner.

Before merging:
- [x] CI needs to pass. Waiting for the PyTorch nightly to include the PR with the above changes.
- [x] Verify Reference Scripts:
  - [x] QAT training
  - [x] PTQ training
  - [x] Accuracy for ResNet18
  - [x] Accuracy for MobileNetV2
  - [x] Accuracy for MobileNetV3 Large model

cc @jerryzh168

### CODE REVIEW

File: references/classification/utils.py [hunk 1]
```diff
@@ -344,7 +344,7 @@ def store_model_weights(model, checkpoint_path, checkpoint_key="model", strict=T
 
         # Quantized Classification
         model = M.quantization.mobilenet_v3_large(pretrained=False, quantize=False)
-        model.fuse_model()
+        model.fuse_model(is_qat=True)
```
Comment (jerryzh168) [inline references/classification/utils.py:347]: would you like to use fuse_model and fuse_model_qat or the flag?

**Review by jerryzh168 — APPROVED** [review]
Looks good I think, thanks for updating the APIs!

File: references/classification/utils.py (hunk 1 above)
Comment (datumbox) [reply references/classification/utils.py:347]: Thanks for the comment. Apologies, I'm not sure I follow. Could you clarify?
